### PR TITLE
[UX][RollingBatch] add details function to the rolling batch

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Support `details:true` in the payload to extract log probability, token id and finish reason

### VLLM JSON

```
curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-typat got Oscar this year","parameters":{"max_new_tokens":50, "do_sample":true, "details":true}}'

{"generated_text": "? I don't understand a single word of it. Lepython knows nothing about movies that don't have subs.

In terms of knowledge, he is a bit like a caged bird. He has a lot of knowledge,", "details": {"finish_reason": "length", "tokens": [{"id": 29973, "text": "?", "log_prob": -4.198227405548096, "special_token": null}, {"id": 306, "text": " I", "log_prob": -6.91507625579834, "special_token": null}, {"id": 1016, "text": " don", "log_prob": -9.707087516784668, "special_token": null}, {"id": 29915, "text": "'", "log_prob": -9.763358518481255, "special_token": null}, {"id": 29873, "text": "t", "log_prob": -9.763435286316962, "special_token": null}, {"id": 2274, "text": " understand", "log_prob": -12.272531908784003, "special_token": null}, {"id": 263, "text": " a", "log_prob": -15.861447256836982, "special_token": null}, {"id": 2323, "text": " single", "log_prob": -17.97309986952314, "special_token": null}, {"id": 1734, "text": " word", "log_prob": -18.216011342439742, "special_token": null}, {"id": 310, "text": " of", "log_prob": -18.690241542492004, "special_token": null}, {"id": 372, "text": " it", "log_prob": -19.81747147146234, "special_token": null}, {"id": 29889, "text": ".", "log_prob": -20.654114213619323, "special_token": null}, {"id": 951, "text": " Le", "log_prob": -29.752767053279968, "special_token": null}, {"id": 4691, "text": "python", "log_prob": -42.31514212194452, "special_token": null}, {"id": 9906, "text": " knows", "log_prob": -49.06187674108514, "special_token": null}, {"id": 3078, "text": " nothing", "log_prob": -51.287539449367614, "special_token": null}, {"id": 1048, "text": " about", "log_prob": -51.431268242035, "special_token": null}, {"id": 2351, "text": " mov", "log_prob": -52.98433199230203, "special_token": null}, {"id": 583, "text": "ies", "log_prob": -52.98517611456191, "special_token": null}, {"id": 393, "text": " that", "log_prob": -57.826722173216695, "special_token": null}, {"id": 1016, "text": " don", "log_prob": -58.956387309553975, "special_token": null}, {"id": 29915, "text": "'", "log_prob": -58.95790488385683, "special_token": null}, {"id": 29873, "text": "t", "log_prob": -58.9599467601729, "special_token": null}, {"id": 505, "text": " have", "log_prob": -59.27860263847833, "special_token": null}, {"id": 11684, "text": " subs", "log_prob": -65.90720562004572, "special_token": null}, {"id": 29889, "text": ".", "log_prob": -66.04373236858373, "special_token": null}, {"id": 13, "text": "\n", "log_prob": -66.79418977462774, "special_token": null}, {"id": 13, "text": "\n", "log_prob": -67.04983835780149, "special_token": null}, {"id": 797, "text": "In", "log_prob": -72.285085969801, "special_token": null}, {"id": 4958, "text": " terms", "log_prob": -77.14496832453733, "special_token": null}, {"id": 310, "text": " of", "log_prob": -77.14528560931649, "special_token": null}, {"id": 7134, "text": " knowledge", "log_prob": -82.65715456302132, "special_token": null}, {"id": 29892, "text": ",", "log_prob": -83.21205973918404, "special_token": null}, {"id": 540, "text": " he", "log_prob": -87.27879214580025, "special_token": null}, {"id": 338, "text": " is", "log_prob": -87.70376959736313, "special_token": null}, {"id": 263, "text": " a", "log_prob": -88.82077232296433, "special_token": null}, {"id": 2586, "text": " bit", "log_prob": -89.70117953712906, "special_token": null}, {"id": 763, "text": " like", "log_prob": -90.28422847683396, "special_token": null}, {"id": 263, "text": " a", "log_prob": -90.62968114311661, "special_token": null}, {"id": 274, "text": " c", "log_prob": -95.71936276848282, "special_token": null}, {"id": 4063, "text": "aged", "log_prob": -99.0848574965712, "special_token": null}, {"id": 11199, "text": " bird", "log_prob": -99.18256937260594, "special_token": null}, {"id": 29889, "text": ".", "log_prob": -100.66095327134099, "special_token": null}, {"id": 940, "text": " He", "log_prob": -101.27128516430821, "special_token": null}, {"id": 756, "text": " has", "log_prob": -102.35117613072362, "special_token": null}, {"id": 263, "text": " a", "log_prob": -102.83534805769887, "special_token": null}, {"id": 3287, "text": " lot", "log_prob": -102.92088392670121, "special_token": null}, {"id": 310, "text": " of", "log_prob": -102.92147074100649, "special_token": null}, {"id": 7134, "text": " knowledge", "log_prob": -103.31259394642984, "special_token": null}, {"id": 29892, "text": ",", "log_prob": -103.76287491080438, "special_token": null}]}}
```

### VLLM JSONLINES

```
root@36a40c9d7904:/opt/djl/test# curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"id": 471, "text": " was", "log_prob": -3.8232274055480957, "special_token": null, "outputs": [" was"]}
{"id": 3721, "text": " exactly", "log_prob": -7.90505838394165, "special_token": null, "outputs": [" exactly"]}
{"id": 1048, "text": " about", "log_prob": -4.2615156173706055, "special_token": null, "outputs": [" about"]}
{"id": 393, "text": " that", "log_prob": -1.0914099216461182, "special_token": null, "outputs": [" that"]}
{"id": 29901, "text": ":", "log_prob": -2.1695868968963623, "special_token": null, "outputs": [":"]}
{"id": 263, "text": " a", "log_prob": -1.5045441389083862, "special_token": null, "outputs": [" a"]}
{"id": 734, "text": " te", "log_prob": -6.200932502746582, "special_token": null, "outputs": [" te"]}
{"id": 264, "text": "en", "log_prob": -0.015936268493533134, "special_token": null, "outputs": ["en"]}
{"id": 482, "text": "age", "log_prob": -0.1848660558462143, "special_token": null, "outputs": ["age"]}
{"id": 7826, "text": " girl", "log_prob": -0.1299373060464859, "special_token": null, "outputs": [" girl"]}
{"id": 1058, "text": " who", "log_prob": -0.5781433582305908, "special_token": null, "outputs": [" who"]}
{"id": 6057, "text": " runs", "log_prob": -3.5494749546051025, "special_token": null, "outputs": [" runs"]}
{"id": 3448, "text": " away", "log_prob": -0.0030378405936062336, "special_token": null, "outputs": [" away"]}
{"id": 515, "text": " from", "log_prob": -0.08086396008729935, "special_token": null, "outputs": [" from"]}
{"id": 3271, "text": " home", "log_prob": -0.2216012328863144, "special_token": null, "outputs": [" home"]}
{"id": 322, "text": " and", "log_prob": -0.5253413319587708, "special_token": null, "outputs": [" and"]}
{"id": 298, "text": " h", "log_prob": -2.0357306003570557, "special_token": null, "outputs": [" h"]}
{"id": 2335, "text": "itch", "log_prob": -0.19932056963443756, "special_token": null, "outputs": ["itch"]}
{"id": 267, "text": "es", "log_prob": -0.12807828187942505, "special_token": null, "outputs": ["es"]}
{"id": 263, "text": " a", "log_prob": -0.041458845138549805, "special_token": null, "outputs": [" a"]}
{"id": 22203, "text": " ride", "log_prob": -0.0012852036161348224, "special_token": null, "outputs": [" ride"]}
{"id": 411, "text": " with", "log_prob": -0.31983014941215515, "special_token": null, "outputs": [" with"]}
{"id": 263, "text": " a", "log_prob": -0.029048359021544456, "special_token": null, "outputs": [" a"]}
{"id": 534, "text": " tr", "log_prob": -0.2750483751296997, "special_token": null, "outputs": [" tr"]}
{"id": 2707, "text": "uck", "log_prob": -0.005570484325289726, "special_token": null, "outputs": ["uck"], "details": {"finish_reason": "length"}}
```

### LMI-DIST JSONLINES

```
root@36a40c9d7904:/opt/djl/test# curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"id": 338, "text": " is", "log_prob": -0.642578125, "special_token": false, "outputs": " is"}
{"id": 376, "text": " \"", "log_prob": -2.26953125, "special_token": false, "outputs": " \""}
{"id": 2177, "text": "Par", "log_prob": -0.16650390625, "special_token": false, "outputs": "Par"}
{"id": 294, "text": "as", "log_prob": -8.416175842285156e-05, "special_token": false, "outputs": "as"}
{"id": 568, "text": "ite", "log_prob": -0.00014400482177734375, "special_token": false, "outputs": "ite"}
{"id": 29908, "text": "\"", "log_prob": -0.263427734375, "special_token": false, "outputs": "\""}
{"id": 10624, "text": " directed", "log_prob": -0.919921875, "special_token": false, "outputs": " directed"}
{"id": 491, "text": " by", "log_prob": -0.0027618408203125, "special_token": false, "outputs": " by"}
{"id": 350, "text": " B", "log_prob": -0.02532958984375, "special_token": false, "outputs": " B"}
{"id": 549, "text": "ong", "log_prob": -0.0004973411560058594, "special_token": false, "outputs": "ong"}
{"id": 3650, "text": " Jo", "log_prob": -0.00981903076171875, "special_token": false, "outputs": " Jo"}
{"id": 265, "text": "on", "log_prob": -0.00019347667694091797, "special_token": false, "outputs": "on"}
{"id": 29899, "text": "-", "log_prob": -0.041656494140625, "special_token": false, "outputs": "-"}
{"id": 1251, "text": "ho", "log_prob": -0.00725555419921875, "special_token": false, "outputs": "ho"}
{"id": 29889, "text": ".", "log_prob": -0.2222900390625, "special_token": false, "outputs": "."}
{"id": 334, "text": " *", "log_prob": -9.453125, "special_token": false, "outputs": " *"}
{"id": 3267, "text": "OS", "log_prob": -3.724609375, "special_token": false, "outputs": "OS"}
{"id": 29907, "text": "C", "log_prob": -0.006988525390625, "special_token": false, "outputs": "C"}
{"id": 1718, "text": "AR", "log_prob": -0.00939178466796875, "special_token": false, "outputs": "AR"}
{"id": 14693, "text": " NE", "log_prob": -6.15625, "special_token": false, "outputs": " NE"}
{"id": 7811, "text": "WS", "log_prob": -0.0025653839111328125, "special_token": false, "outputs": "WS"}
{"id": 14445, "text": " AL", "log_prob": -1.791015625, "special_token": false, "outputs": " AL"}
{"id": 20161, "text": "ERT", "log_prob": -0.0002162456512451172, "special_token": false, "outputs": "ERT"}
{"id": 29930, "text": "*", "log_prob": -0.020416259765625, "special_token": false, "outputs": "*"}
{"id": 334, "text": " *", "log_prob": -2.958984375, "special_token": false, "outputs": " *", "details": {"finish_reason": "length"}}
```

### LMI-DIST JSON

```
root@36a40c9d7904:/opt/djl/test# curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"generated_text": " is SPARE PARTS. Here is the first look of the new trailer.

"The wh", "details": {"finish_reason": "length", "tokens": [{"id": 338, "text": " is", "log_prob": -0.642578125, "special_token": false}, {"id": 317, "text": " S", "log_prob": -3.9296875, "special_token": false}, {"id": 7228, "text": "PA", "log_prob": -3.072265625, "special_token": false}, {"id": 1525, "text": "RE", "log_prob": -0.1630859375, "special_token": false}, {"id": 349, "text": " P", "log_prob": -0.78173828125, "special_token": false}, {"id": 8322, "text": "ART", "log_prob": -0.0025959014892578125, "special_token": false}, {"id": 29903, "text": "S", "log_prob": -0.046417236328125, "special_token": false}, {"id": 29889, "text": ".", "log_prob": -0.6396484375, "special_token": false}, {"id": 2266, "text": " Here", "log_prob": -4.07421875, "special_token": false}, {"id": 338, "text": " is", "log_prob": -0.481689453125, "special_token": false}, {"id": 278, "text": " the", "log_prob": -0.62548828125, "special_token": false}, {"id": 29871, "text": " ", "log_prob": -5.125, "special_token": false}, {"id": 937, "text": "first", "log_prob": -3.201171875, "special_token": false}, {"id": 1106, "text": " look", "log_prob": -1.2158203125, "special_token": false}, {"id": 310, "text": " of", "log_prob": -1.130859375, "special_token": false}, {"id": 278, "text": " the", "log_prob": -0.2088623046875, "special_token": false}, {"id": 716, "text": " new", "log_prob": -2.40234375, "special_token": false}, {"id": 1020, "text": " tra", "log_prob": -2.005859375, "special_token": false}, {"id": 3955, "text": "iler", "log_prob": -0.0016698837280273438, "special_token": false}, {"id": 29889, "text": ".", "log_prob": -0.88623046875, "special_token": false}, {"id": 13, "text": "\n", "log_prob": -0.438720703125, "special_token": false}, {"id": 13, "text": "\n", "log_prob": -0.90185546875, "special_token": false}, {"id": 29908, "text": "\"", "log_prob": -3.03515625, "special_token": false}, {"id": 1576, "text": "The", "log_prob": -2.169921875, "special_token": false}, {"id": 377, "text": " wh", "log_prob": -8.1875, "special_token": false}]}}
```

### DeepSpeed JSON

```
root@36a40c9d7904:/opt/djl/test# curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"generated_text": " is the first movie to reach the viewer in a meaningful way, the realm of  the film gives a sense", "details": {"finish_reason": "length", "tokens": [{"id": 338, "text": " is", "log_prob": -0.6295561194419861, "special_token": false}, {"id": 278, "text": " the", "log_prob": -2.8583195209503174, "special_token": false}, {"id": 937, "text": " first", "log_prob": -3.3326945304870605, "special_token": false}, {"id": 14064, "text": " movie", "log_prob": -1.5597784519195557, "special_token": false}, {"id": 304, "text": " to", "log_prob": -1.3934849500656128, "special_token": false}, {"id": 6159, "text": " reach", "log_prob": -4.0857648849487305, "special_token": false}, {"id": 278, "text": " the", "log_prob": -2.124640464782715, "special_token": false}, {"id": 6316, "text": " vie", "log_prob": -10.708840370178223, "special_token": false}, {"id": 556, "text": "wer", "log_prob": -0.0009380945703014731, "special_token": false}, {"id": 297, "text": " in", "log_prob": -1.8945964574813843, "special_token": false}, {"id": 263, "text": " a", "log_prob": -0.898783802986145, "special_token": false}, {"id": 6593, "text": " meaning", "log_prob": -9.064022064208984, "special_token": false}, {"id": 1319, "text": "ful", "log_prob": -0.024169210344552994, "special_token": false}, {"id": 982, "text": " way", "log_prob": -0.017020229250192642, "special_token": false}, {"id": 29892, "text": ",", "log_prob": -1.5901317596435547, "special_token": false}, {"id": 278, "text": " the", "log_prob": -1.907080888748169, "special_token": false}, {"id": 1855, "text": " real", "log_prob": -5.575304985046387, "special_token": false}, {"id": 29885, "text": "m", "log_prob": -2.956176996231079, "special_token": false}, {"id": 310, "text": " of", "log_prob": -0.07764696329832077, "special_token": false}, {"id": 29871, "text": " ", "log_prob": -4.187904357910156, "special_token": false}, {"id": 278, "text": " the", "log_prob": -2.589885950088501, "special_token": false}, {"id": 2706, "text": " film", "log_prob": -3.337740421295166, "special_token": false}, {"id": 4076, "text": " gives", "log_prob": -6.683914661407471, "special_token": false}, {"id": 263, "text": " a", "log_prob": -1.3925080299377441, "special_token": false}, {"id": 4060, "text": " sense", "log_prob": -3.5028796195983887, "special_token": false}]}}
```

### DeepSpeed JSONLINES

```
root@36a40c9d7904:/opt/djl/test# curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"id": 338, "text": " is", "log_prob": -0.6295561194419861, "special_token": false, "outputs": " is"}
{"id": 278, "text": " the", "log_prob": -2.8583195209503174, "special_token": false, "outputs": " the"}
{"id": 937, "text": " first", "log_prob": -3.3326945304870605, "special_token": false, "outputs": " first"}
{"id": 14064, "text": " movie", "log_prob": -1.5597784519195557, "special_token": false, "outputs": " movie"}
{"id": 304, "text": " to", "log_prob": -1.3934849500656128, "special_token": false, "outputs": " to"}
{"id": 6159, "text": " reach", "log_prob": -4.0857648849487305, "special_token": false, "outputs": " reach"}
{"id": 278, "text": " the", "log_prob": -2.124640464782715, "special_token": false, "outputs": " the"}
{"id": 6316, "text": " vie", "log_prob": -10.708840370178223, "special_token": false, "outputs": " vie"}
{"id": 556, "text": "wer", "log_prob": -0.0009380945703014731, "special_token": false, "outputs": "wer"}
{"id": 297, "text": " in", "log_prob": -1.8945964574813843, "special_token": false, "outputs": " in"}
{"id": 263, "text": " a", "log_prob": -0.898783802986145, "special_token": false, "outputs": " a"}
{"id": 6593, "text": " meaning", "log_prob": -9.064022064208984, "special_token": false, "outputs": " meaning"}
{"id": 1319, "text": "ful", "log_prob": -0.024169210344552994, "special_token": false, "outputs": "ful"}
{"id": 982, "text": " way", "log_prob": -0.017020229250192642, "special_token": false, "outputs": " way"}
{"id": 29892, "text": ",", "log_prob": -1.5901317596435547, "special_token": false, "outputs": ","}
{"id": 278, "text": " the", "log_prob": -1.907080888748169, "special_token": false, "outputs": " the"}
{"id": 1855, "text": " real", "log_prob": -5.575304985046387, "special_token": false, "outputs": " real"}
{"id": 29885, "text": "m", "log_prob": -2.956176996231079, "special_token": false, "outputs": "m"}
{"id": 310, "text": " of", "log_prob": -0.07764696329832077, "special_token": false, "outputs": " of"}
{"id": 29871, "text": " ", "log_prob": -4.187904357910156, "special_token": false, "outputs": " "}
{"id": 278, "text": " the", "log_prob": -2.589885950088501, "special_token": false, "outputs": " the"}
{"id": 2706, "text": " film", "log_prob": -3.337740421295166, "special_token": false, "outputs": " film"}
{"id": 4076, "text": " gives", "log_prob": -6.683914661407471, "special_token": false, "outputs": " gives"}
{"id": 263, "text": " a", "log_prob": -1.3925080299377441, "special_token": false, "outputs": " a"}
{"id": 4060, "text": " sense", "log_prob": -3.5028796195983887, "special_token": false, "outputs": " sense", "details": {"finish_reason": "length"}}
```

### CODE REVIEW

File: engines/python/setup/djl_python/rolling_batch/rolling_batch.py [hunk 1]
```diff
@@ -64,29 +100,45 @@ def __init__(self, id: int, input_text: str, parameters: dict):
...
+        self.token_cache = None
+        if parameters.get("details", False):
+            self.token_cache = []
 
     def __repr__(self):
         return f"<Request id: {self.id} Input {self.input_text} Parameters {self.parameters} Finished {self.last_token}>"
 
     def set_next_token(self,
-                       next_token: str,
+                       next_token: Union[Token, str],
```
Comment (xyang16) [inline engines/python/setup/djl_python/rolling_batch/rolling_batch.py:114]: This is a Union, not Token object?

Comment (lanking520) [reply engines/python/setup/djl_python/rolling_batch/rolling_batch.py:114]: https://stackoverflow.com/questions/48709104/how-do-i-specify-multiple-types-for-a-parameter-using-type-hints

**Comment by lanking520** [comment]
fixes #1349

**Review by frankfliu — APPROVED** [review]

**Comment by lanking520** [comment]
Updated UX:

For JSONLines

```
ubuntu@ip-172-31-57-107:~$ curl -X POST http://127.0.0.1:8080/invocations -N   -H "Content-type: application/json"   -d '{"inputs":"The new movie that got Oscar this year","parameters":{"max_new_tokens":25, "do_sample":true, "details":true}}'
{"token": {"id": 29892, "text": ",", "log_prob": -2.1982274055480957, "special_token": null}, "details": null, "outputs": [","]}
{"token": {"id": 29871, "text": " ", "log_prob": -5.1034088134765625, "special_token": null}, "details": null, "outputs": [" "]}
{"token": {"id": 322, "text": " and", "log_prob": -4.402134895324707, "special_token": null}, "details": null, "outputs": [" and"]}
{"token": {"id": 896, "text": " they", "log_prob": -4.968091011047363, "special_token": null}, "details": null, "outputs": [" they"]}
{"token": {"id": 1497, "text": " said", "log_prob": -2.523975133895874, "special_token": null}, "details": null, "outputs": [" said"]}
{"token": {"id": 540, "text": " he", "log_prob": -4.212726593017578, "special_token": null}, "details": null, "outputs": [" he"]}
{"token": {"id": 6416, "text": " fall", "log_prob": -8.792691230773926, "special_token": null}, "details": null, "outputs": [" fall"]}
{"token": {"id": 1623, "text": " down", "log_prob": -3.976060390472412, "special_token": null}, "details": null, "outputs": [" down"]}
{"token": {"id": 1156, "text": " after", "log_prob": -4.689115524291992, "special_token": null}, "details": null, "outputs": [" after"]}
{"token": {"id": 13748, "text": " drink", "log_prob": -2.4976370334625244, "special_token": null}, "details": null, "outputs": [" drink"]}
{"token": {"id": 292, "text": "ing", "log_prob": -0.4084559679031372, "special_token": null}, "details": null, "outputs": ["ing"]}
{"token": {"id": 2086, "text": " too", "log_prob": -1.1722670793533325, "special_token": null}, "details": null, "outputs": [" too"]}
{"token": {"id": 1568, "text": " much", "log_prob": -0.008474932983517647, "special_token": null}, "details": null, "outputs": [" much"]}
{"token": {"id": 29889, "text": ".", "log_prob": -2.0607433319091797, "special_token": null}, "details": null, "outputs": ["."]}
{"token": {"id": 13, "text": "\n", "log_prob": -1.797961950302124, "special_token": null}, "details": null, "outputs": ["\n"]}
{"token": {"id": 13, "text": "\n", "log_prob": -0.7299792170524597, "special_token": null}, "details": null, "outputs": ["\n"]}
{"token": {"id": 29899, "text": "-", "log_prob": -2.6099185943603516, "special_token": null}, "details": null, "outputs": ["-"]}
{"token": {"id": 306, "text": " I", "log_prob": -1.9703612327575684, "special_token": null}, "details": null, "outputs": [" I"]}
{"token": {"id": 1348, "text": " think", "log_prob": -0.5400672554969788, "special_token": null}, "details": null, "outputs": [" think"]}
{"token": {"id": 393, "text": " that", "log_prob": -2.418488025665283, "special_token": null}, "details": null, "outputs": [" that"]}
{"token": {"id": 1795, "text": " might", "log_prob": -5.905703544616699, "special_token": null}, "details": null, "outputs": [" might"]}
{"token": {"id": 367, "text": " be", "log_prob": -0.09645677357912064, "special_token": null}, "details": null, "outputs": [" be"]}
{"token": {"id": 263, "text": " a", "log_prob": -0.11685966700315475, "special_token": null}, "details": null, "outputs": [" a"]}
{"token": {"id": 16558, "text": " rum", "log_prob": -1.9316766262054443, "special_token": null}, "details": null, "outputs": [" rum"]}
{"token": {"id": 272, "text": "or", "log_prob": -0.004645507782697678, "special_token": null}, "details": {"finish_reason": "length"}, "outputs": ["or"]}
```

**Pull request merged**